lib/class/json_data.class.php

```diff
@@ -0,0 +1,638 @@
+     * set_offset
+     *
+     * This takes an int and changes the offset
+     *
+     * @param    integer    $offset    (description here...)
+     * @return    void
+     */
+    public static function set_offset($offset)
+    {
+        $offset       = intval($offset);
+        self::$offset = $offset;
+    } // set_offset
+
+    /**
+     * set_limit
+     *
+     * This sets the limit for any ampache transactions
+     *
+     * @param    integer    $limit    (description here...)
+     * @return    void
+     */
+    public static function set_limit($limit)
+    {
+        if (!$limit) {
+            return false;
+        }
+
+        $limit       = intval($limit);
+        self::$limit = $limit;
+    } // set_limit
+
+    /**
+     * error
+     *
+     * This generates a JSON Error message
+     * nothing fancy here...
+     *
+     * @param    integer    $code    Error code
+     * @param    string    $string    Error message
+     * @return    string    return error message JSON
+     */
+    public static function error($code,$string)
+    {
+        $JSON = json_encode(array("error" => array("code" => $code, "message" => $string)), JSON_PRETTY_PRINT);
+        return $JSON;
+    } // error
+
+    /**
+     * single_string
+     *
+     * This takes two values, first the key second the string
+     *
+     * @param    string    $key    (description here...)
+     * @param    string    $string    JSON data
+     * @return    string    return JSON
+     */
+    public static function single_string($key, $string='')
+    {
+        if (!empty($string)) {
+            $JSON = json_encode(array($key => $string), JSON_PRETTY_PRINT);
+        } else {
+            $JSON = json_encode(array("message" => $key), JSON_PRETTY_PRINT);
+        }
+
+        return $JSON;
+    } // single_string
+
+    /**
+     * tags_string
+     *
+     * This returns the formatted 'tags' string for an JSON document
+     *
+     */
+    private static function tags_string($tags)
+    {
+        $JSON = null;
+
+        if (is_array($tags)) {
+            $atags = array();
+            foreach ($tags as $tag_id => $data) {
+                if (array_key_exists($data['id'], $atags)) {
+                    $atags[$data['id']]['count']++;
+                } else {
+                    $atags[$data['id']] = array('name' => $data['name'],
+                        'count' => 1);
+                }
+            }
+
+            foreach ($atags as $id => $data) {
+                $JSON['id'] = $id;
+                $JSON['count'] = $data['count'];
+                $JSON['name'] = $data['name'];
+            }
+        }
+
+        return $JSON;
+    } // tags_string
+
+
+    /**
+     * playlist_song_tracks_string
+     *
+     * This returns the formatted 'playlistTrack' string for an JSON document
+     *
+     */
+    private static function playlist_song_tracks_string($song, $playlist_data)
+    {
+        if (empty($playlist_data)) {
+            return "";
```

Comment:
This will be a duplicate with the `return` on line 165, making the flow hard to follow.

I would write
```php
$playlist_track = "";

foreach ($playlist_data as $playlist) {
    if ($playlist["object_id"] == $song->id) {
        $playlist_track = $playlist["track"];
        break;
    }
}
return $playlist_track;
```

with a single exit point from the function.